src/cascadia/TerminalApp/Pane.h

```diff
@@ -19,140 +19,169 @@
+    //   have 1x1 character visible, in each of its children. If we're a leaf, we'll
+    //   include the space needed for borders _within_ us.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - The minimum size that this pane can be resized to and still have a visible
+    //   character.
+    virtual winrt::Windows::Foundation::Size _GetMinSize() const = 0;
 
-    bool _Resize(const winrt::TerminalApp::Direction& direction);
-    bool _NavigateFocus(const winrt::TerminalApp::Direction& direction);
+    // Method Description:
+    // - Builds a tree of LayoutSizeNode that matches the tree of panes. Each node
+    //   has minimum size that the corresponding pane can have.
+    // Arguments:
+    // - widthOrHeight: if true operates on width, otherwise on height
+    // Return Value:
+    // - Root node of built tree, which matches this pane.
+    virtual LayoutSizeNode _CreateMinSizeTree(const bool widthOrHeight) const;
 
-    void _CloseChild(const bool closeFirst);
+    // Method Description:
+    // - Adjusts given dimension (width or height) so that all descendant terminals
+    //   align with their character grids as close as possible. Also makes sure to
+    //   fit in minimal sizes of the panes.
+    // Arguments:
+    // - widthOrHeight: if true operates on width, otherwise on height
+    // - dimension: a dimension (width or height) to be snapped
+    // Return Value:
+    // - pair of floats, where first value is the size snapped downward (not greater then
+    //   requested size) and second is the size snapped upward (not lower than requested size).
+    //   If requested size is already snapped, then both returned values equal this value.
+    virtual SnapSizeResult _CalcSnappedDimension(const bool widthOrHeight, const float dimension) const = 0;
+
+    // Method Description:
+    // - Increases size of given LayoutSizeNode to match next possible 'snap'. In case of leaf
+    //   pane this means the next cell of the terminal. Otherwise it means that one of its children
+    //   advances (recursively). It expects the given node and its descendants to have either
+    //   already snapped or minimum size.
+    // Arguments:
+    // - widthOrHeight: if true operates on width, otherwise on height.
+    // - sizeNode: a layouting node that corresponds to this pane.
+    // Return Value:
+    // - <none>
+    virtual void _AdvanceSnappedDimension(const bool widthOrHeight, LayoutSizeNode& sizeNode) const = 0;
 
-    void _FocusFirstChild();
-    void _ControlConnectionStateChangedHandler(const winrt::Microsoft::Terminal::TerminalControl::TermControl& sender, const winrt::Windows::Foundation::IInspectable& /*args*/);
+    // Make derived classes friends, so they can access protected members of Pane. C++ for some
+    // reason doesn't allow that by default - e.g. LeafPane can access LeafPane::_ProtectedMember,
+    // but not Pane::_ProtectedMember.
+    friend class LeafPane;
+    friend class ParentPane;

```

Comment:
I'm uneasy about this. I feel like there is a better way around this.

Reply:
Ok, from some other major programming languages I checked, only Java allows that (to access protected member of base class inside the scope of derived class). That said, I still believe this is the correct/most appropriate way of doing things here.